libpod/in_memory_state.go

```diff
@@ -303,6 +293,86 @@ func (s *InMemoryState) RemovePod(pod *Pod) error {
 	return nil
 }
 
+// UpdatePod updates a pod's state from the backing database
+// As in-memory states have no database this is a no-op
+func (s *InMemoryState) UpdatePod(pod *Pod) error {
+	return nil
+}
+
+// AddContainerToPod adds a container to the given pod, also adding it to the
+// state
+func (s *InMemoryState) AddContainerToPod(pod *Pod, ctr *Container) error {
+	if !pod.valid {
+		return errors.Wrapf(ErrPodRemoved, "pod %s is not valid and cannot be added to", pod.ID())
+	} else if !ctr.valid {
+		return errors.Wrapf(ErrCtrRemoved, "container %s is not valid and cannot be added to the pod", ctr.ID())
+	}
+
+	if ctr.config.Pod != pod.ID() {
```

Comment:
This seems wierd.  Shouldn't we have a pod object where we are adding a container
pod.Add(ctr)

I would think this is a case where the error would be ctr is already in a different pod.  pod.Add() below should return that error and this check should not be made?

Reply:
We handle the pod.addContainer a little lower, on like 340. By forcing it to go through the state, we make sure the container is added to the DB at the same time if we have one.